### ISSUE TITLE
Code Sniffer Error

### ISSUE THREAD

**Issue by gianluca78** [nature/pa11y#68]
Hello!
I got a generic "Code Sniffer Error" when I launch the following command:
    pa11y scuolafalconeborsellino.it

Also when  use the flag -d, I can't understand what went wrong:

>Welcome to Pa11y
>We'll sniff your page for you now.

> > Debug: Starting timeout timer (30000ms)
> > Loading page...
 > Running HTML CodeSniffer...
 > Error: HTML CodeSniffer error

What to do to better understand the problem? I suppose this is caused by the frame included in the webpage...

Thank you very much!

**Comment by whymarrh** [comment]
What version of PhantomJS are you using? (And, just for good measure, what version of pa11y?)

When I run `pa11y scuolafalconeborsellino.it`:

```
$ phantomjs --version
1.9.8
$ pa11y scuolafalconeborsellino.it

Welcome to Pa11y
We'll sniff your page for you now.

 > Loading page...
phantom stdout: TypeError: 'undefined' is not a function (evaluating 'container.avAddShareTrackingEvents()')


phantom stdout:   http://scuolafalconeborsellino.altervista.org/_av_share/static/wordpress/js/social.js?ver=2015.05.06.17.45:151
  http://scuolafalconeborsellino.altervista.org/_av_share/static/wordpress/js/social.js?ver=2015.05.06.17.45:153

 > Running HTML CodeSniffer...
 > Error: HTML CodeSniffer error


```

I would try (if possible) to fix the errors shown and see if PhantomJS can load the page correctly.

**Comment by gianluca78** [comment]
Thank you. I use PhantomJs 1.9.8 on Ubuntu 14.04 and pa11y 1.6.3.

Unfortunately I can't edit the website. It is not mine.

It could be useful to see the same log of you. I'm evaluating an array of websites with pa11y and when CodeSniffer doesn't complete its task, an empty report file is generated. I would like to be able to distinguish if a report is empty because there are not errors or because CodeSniffer had a problem...

**Comment by whymarrh** [comment]
If you are using just the CLI, you can know that there were errors from pa11y's exit code ([L116-L118](https://github.com/nature/pa11y/blob/7d1cc22fd98fad44a3a6c5c37e6b5a62623f47c3/bin/pa11y#L116-L118)).

**Comment by gianluca78** [comment]
No, I use it from a R script. By the way, I tried:

```
pa11y www.google.it
echo $?
```
And the result is 1.

With the scuolafalconeborsellino website:
```
pa11y scuolafalconeborsellino.it
echo $?
```

The result is 255 despite the CodeSniffer error.

Am I doing something wrong? Thanks!

**Comment by whymarrh** [comment]
@gianluca78 you should be able to express the return code as its original value. An example with Perl:

```
$ pa11y scuolafalconeborsellino.it
$ perl -e "print unpack('c', pack('C', $?)), \$/"
```

You should be able to get the same functionality in R with the [`pack`](http://cran.r-project.org/web/packages/pack/index.html) package.

With that said, in thinking about this now, this functionality (the use of a negative error code, introduced in #32) isn't POSIX compliant and will break in unfavourable ways if there are too many errors on the page.

**Comment by rowanmanning** [comment]
Hi, as well as the issues @whymarrh has raised, I think some general tweaking needs to happen around the error reporting – it's not very verbose at the moment and it'd be useful to get a full error stack. I'll work on this as well.

**Comment by rowanmanning** [comment]
This should be fixed in pa11y 2.0, please re-open if you continue to have issues
